### DISCUSSION TITLE
Blinding

### DISCUSSION THREAD

**Discussion by rkansal47** [scikit-hep/coffea#506] [Q&A] [answered]
Hi, is there a recommended way of blinding histogram plots? i.e. is there a way to perhaps remove specific non-sparse bins, or set the values of specific bins to 0 after filling a histogram?

**Comment by lgray** [comment]
I think this can be achieved with a few lines, scale the histograms to zero.
If you need to remove the counts and other information that's possible too.
Maybe what we need to do on the coffea side is add something that returns an empty copy of a (collection of) histograms?

assuming you have a list of histograms, or a dict:
```python3
raw_hists = {'hist1': stuff, 'hist2' ...}
blinded_hists = deepcopy(raw_hists)
for hist in blinded_hists:
    hist.scale(0.)
# then go on to plot blinded_hists
```

You could stick the above in a function, control it with a bool so you can unblind easily later.

@nsmith- your thoughts?

**Reply by nsmith-** [reply]
Wouldn't that just set all bins to zero? I think here the desire is to set specific bins to zero.

**Reply by lgray** [reply]
Ah yeah, so really is there a case for a new verb in our package / maybe hist?=

**Comment by nsmith-** [comment, marked as answer]
With coffea hists:
```python
import coffea.hist
import numpy as np

h = coffea.hist.Hist("Events", coffea.hist.Bin("x", "axis", 10, 0, 1))
h.fill(x=np.random.normal(loc=0.5, scale=0.2, size=1000))

# blind bins 2 and 3
h.values()[()][2:4] = 0

coffea.hist.plot1d(h)
```
producing
![image](https://user-images.githubusercontent.com/6587412/115778752-bdc4de00-a37c-11eb-93bb-6643a4b1e933.png)


With [hist](https://github.com/scikit-hep/hist):
```python
import hist
import numpy as np

h = (
    hist.Hist.new.Reg(10, 0, 1, name="x", label="axis")
    .Weight()
    .fill(x=np.random.normal(loc=0.5, scale=0.2, size=1000))
)

# blind bins 2 and 3
h.view().value[2:4] = 0

h.plot1d()
```
producing
![image](https://user-images.githubusercontent.com/6587412/115778849-db924300-a37c-11eb-9272-839e6291d295.png)

**Reply by nsmith-** [reply]
Can add `yerr=True` to the hist plot1d to get the errorbars.
